src/bsp/platform/posix/xi_bsp_fwu_posix.c

```diff
@@ -45,21 +96,42 @@ void xi_bsp_fwu_on_package_download_finished( const char* const firmware_resourc
 {
     printf( "--- %s, firmware name: %s\n", __FUNCTION__, firmware_resource_name );
 
-    // - rename file to firmware_resource_name ## _DATETIMESTAMP
+    if ( NULL == firmware_resource_name )
+    {
+        return;
+    }
+
+    /* - rename file to firmware_resource_name ## _DATETIMESTAMP */
     const char* update_fw_name = _generate_update_filename( firmware_resource_name );
     rename( firmware_resource_name, update_fw_name );
 
 #if 0
-    // - start new firmware
+    {
+        /* - save the new firmware progname to identify new firmware first run
+         *   in function xi_bsp_fwu_on_new_firmware_ok */
+
+        xi_bsp_io_fs_resource_handle_t resource_handle =
+            XI_BSP_IO_FS_INVALID_RESOURCE_HANDLE;
+
+        const char* firmware_test_flag_filename =
+            _get_firmware_test_flag_filename( update_fw_name );
+
+        xi_bsp_io_fs_open( firmware_test_flag_filename, 0, XI_BSP_IO_FS_OPEN_WRITE,
+                           &resource_handle );
+
+        xi_bsp_io_fs_close( resource_handle );
+    }
+
+    /* - start new firmware */
     char system_command[512] = {0};
 
     sprintf( system_command, "chmod a+x %s; "
                              "./%s "
-                             "-a 4d3c7986-8d53-4cf8-903e-7fd30ff63be1 "
-                             "-u fc4ba77e-0ba0-4ef2-bb46-d30bc59546ab "
-                             "-P JMnzPt3zDH2ZY2Q/ZQRgzublMGF6VfAFv7vFxFf5tLY= "
-                             "-p xi/blue/v1/4d3c7986-8d53-4cf8-903e-7fd30ff63be1/d/"
-                             "b03e863d-2478-48cf-976e-fa2d5ddf4ec8/Temperature",
+                             "-a <Xively Account ID> "
+                             "-u <Xively Device ID> "
+                             "-P <Xively Device Password> "
+                             "-p xi/blue/v1/<Xively Account ID>/d/"
+                             "<Xively Device ID>/Temperature",
```

Comment:
Hold on, writing these down...

Reply:
oh so we DO still need the account id?

Reply:
seems so, yes. Maybe control topic does not need accountid, but apparently "other" topic names include it.

Reply:
but still the xi_initialize needs the username despite it is passed in xi_connect too later.